io.lmms.LMMS.json

```diff
@@ -0,0 +1,123 @@
+{
+  "app-id": "io.lmms.LMMS",
+  "runtime": "org.kde.Platform",
+  "runtime-version": "5.12",
+  "sdk": "org.kde.Sdk",
+  "command": "lmms",
+  "rename-desktop-file": "lmms.desktop",
+  "rename-appdata-file": "lmms.appdata.xml",
+  "rename-icon": "lmms",
+  "finish-args": [
+    /* X11 + XShm access */
+    "--share=ipc",
+    "--socket=x11",
+    /* Wayland */
+    "--socket=wayland",
+    /* Note playback */
+    "--socket=pulseaudio",
+    /* MIDI */
+    "--device=all",
+    /* Allow loading, saving files from anywhere (portals don’t work yet) */
+    "--filesystem=host",
+    /* Allow other instances to see lockfiles */
+    "--env=TMPDIR=/var/tmp",
+    "--env=QT_AUTO_SCREEN_SCALE_FACTOR=1"
+  ],
+  "cleanup": [
+    "/include",
+    "/man",
+    "/share/man",
+    "/lib/pkgconfig",
+    "*.la"
+  ],
+  "modules": [
+    {
+      "name": "fftw3",
+      "config-opts": [
+        "--enable-threads",
+        "--enable-shared",
+        "--enable-float",
+        "--disable-static"
+      ],
+      "build-options": {
+        "arch": {
+          "x86_64": {
+            "config-opts": [
+              "--enable-sse2",
+              "--enable-avx",
+              "--enable-avx-128-fma"
+            ]
+          }
+        }
+      },
+      "sources": [
+        {
+          "type": "archive",
+          "url": "http://www.fftw.org/fftw-3.3.7.tar.gz",
+          "sha256": "3b609b7feba5230e8f6dd8d245ddbefac324c5a6ae4186947670d9ac2cd25573"
+        }
+      ],
+      "cleanup": [
+        "/bin",
+        "/include",
+        "/lib/*.la"
+      ]
+    },
+    {
+      "name": "portaudio",
+      "config-opts": [
+        "--disable-static",
+        "--without-oss"
+      ],
+      "sources": [
+        {
+          "type": "archive",
+          "url": "http://www.portaudio.com/archives/pa_stable_v190600_20161030.tgz",
+          "sha256": "f5a21d7dcd6ee84397446fa1fa1a0675bb2e8a4a6dceb4305a8404698d8d1513"
+        }
+      ]
+    },
+    {
+      "name": "fluidsynth",
+      "buildsystem": "cmake-ninja",
+      "config-opts": [
+        "-DCMAKE_BUILD_TYPE=RelWithDebInfo",
+        "-DLIB_SUFFIX=",
+        "-DCMAKE_EXE_LINKER_FLAGS=-lncurses",
+        "-DCMAKE_SHARED_LINKER_FLAGS=-lncurses",
+        "-Denable-ladspa=on",
+        "-Denable-jack=off"
+      ],
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://github.com/FluidSynth/fluidsynth/archive/v1.1.11.tar.gz",
+          "sha256": "da8878ff374d12392eecf87e96bad8711b8e76a154c25a571dd8614d1af80de8"
+        }
+      ]
+    },
+    {
+      "name": "lmms",
+      "buildsystem": "cmake-ninja",
+      "config-opts": [
+        "-DWANT_QT5=1"
+      ],
+      "post-install": [
+        "mkdir -p /app/share/appdata",
+        "install -m644 -t /app/share/appdata io.lmms.LMMS.appdata.xml"
+      ],
+      "sources": [
+        {
+          "type": "git",
+          "url": "https://github.com/LMMS/lmms.git",
+          "branch": "stable-1.2",
```

Comment:
As far as I know, specifying the `branch` is unnecessary and can be harmful. Flatpak assumes that the given `commit` is the top commit of this branch. In the other words, build will fail if vendor updates the branch. Providing a `tag` is a much better idea.

Reply:
Tag `v1.2.0-rc7` is current "stable".  `v1.2.0-rc8` will be released very soon.

Reply:
Personally I'd just leave it at commit only, but YMMV, tag is fine. The important point is that it has to be reproducible and specific.